### PR TITLE
SNAP-947 On enter trigger search on dob field

### PR DESCRIPTION
**Pull request by pokharelmanish**
### Jira Story

- [Trigger search when enter is pressed in Dob field](https://osi-cwds.atlassian.net/browse/SNAP-947)

## Description
Trigger search when enter is pressed in Dob field.
Note: This is part2 of SNAP-947. 
           Part1: https://github.com/ca-cwds/intake/pull/1094

## Tests
- [x] I have included unit tests 
- [ ] I have included feature tests 
- [ ] I have included other tests 
- [ ] I have NOT included tests 

## Types of changes

- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (No behavioral changes)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] I have ran all tests for the project.
- [ ] I have ran lint/rubocop check for the changed/new files.
- [ ] My code is in a stable state ready to be deployable, but not necessarily complete.
- [ ] I promise on my honor as a CWDS developer to ensure I don't break the build, to check Lint/Rubocop, use best practices, to leave the code in better shape than I found it, and to have a good day.

### CODE REVIEW

**Comment by pokharelmanish** [comment]
retest this please

File: app/javascript/common/search/Autocompleter.jsx [hunk 1]
```diff
@@ -201,13 +198,15 @@ export default class Autocompleter extends Component {
...
+
   renderPersonSearchFields() {
     const {states, counties, onChange, onCancel, onClear, onBlur, personSearchFields, isAdvancedSearchOn, clientIdError, ssnErrors, dobErrors, canSearch} = this.props
     const searchWithEnter = (e) => {
       const enterKeyCode = 13
-      if (canSearch && e.charCode === enterKeyCode) { return this.handleSubmit() }
-      return null
+      if ((canSearch && e.charCode === enterKeyCode)) { this.handleSubmit() }
     }
+    const dobSearchWithEnter = (e) => { if (this.isValidDate(e)) { onChange('searchDateOfBirth', moment(e.target.value).format('YYYY-MM-DD')) } }
```
Comment (ggomez-cambriasolutions) [inline app/javascript/common/search/Autocompleter.jsx]: I think this should have a different name. We are validating a date, and setting redux with the value. I don't see any event listeners for the enter key in this method.

Comment (ggomez-cambriasolutions) [reply app/javascript/common/search/Autocompleter.jsx]: maybe `validateAndSetDateOfBirth`?

Comment (pokharelmanish) [reply app/javascript/common/search/Autocompleter.jsx]: 👍

**Pull request merged**